test/common_utils.py

```diff
@@ -270,84 +270,57 @@ def combinations_grid(**kwargs):
     return [dict(zip(kwargs.keys(), values)) for values in itertools.product(*kwargs.values())]
 
 
-class ImagePair(TensorLikePair):
-    def __init__(
-        self,
-        actual,
-        expected,
-        *,
-        mae=False,
-        **other_parameters,
-    ):
-        if all(isinstance(input, PIL.Image.Image) for input in [actual, expected]):
-            actual, expected = [to_image_tensor(input) for input in [actual, expected]]
+assert_equal = functools.partial(torch.testing.assert_close, rtol=0, atol=0)
 
-        super().__init__(actual, expected, **other_parameters)
-        self.mae = mae
 
-    def compare(self) -> None:
-        actual, expected = self.actual, self.expected
+def assert_close_with_image_support(actual, expected, *, mae=False, atol=None, msg=None, **kwargs):
+    def compare(actual, expected):
+        if all(isinstance(input, PIL.Image.Image) for input in [actual, expected]):
+            actual, expected = [to_image_tensor(input) for input in [actual, expected]]
 
-        self._compare_attributes(actual, expected)
-        actual, expected = self._equalize_attributes(actual, expected)
+        enable_mae_comparison = all(isinstance(input, torch.Tensor) for input in [actual, expected])
+
+        value_comparison_failure = False
+
+        def msg_callback(default_msg):
+            # This is a dirty hack that let's us "hook" into the comparison logic of `torch.testing.assert_close`. It
+            # will only be triggered in case of failed value comparison. This let's us reuse all of the attribute
+            # checking that `torch.testing.assert_close` does while giving us the ability to ignore the regular value
+            # comparison.
```

Comment:
Per comment. Note while this is a hack, it doesn't depend on any non-public functionality or assumptions.

Reply:
NIT
```suggestion
            # This is a dirty hack that lets us "hook" into the comparison logic of `torch.testing.assert_close`. It
            # will only be triggered in case of failed value comparison. This lets us reuse all of the attribute
            # checking that `torch.testing.assert_close` does while giving us the ability to ignore the regular value
            # comparison.
```